clamav/client.py

```diff
@@ -61,3 +66,37 @@ def health(self):
         url = self.routes.health()
         response = self._request(self._session.get, url)
         return ClamAVHealth(response.json())
+
+    def scan_container_image(self, image_reference: str) -> typing.Tuple[ClamAVScanResult, str]:
+        '''Fetch and scan the container image with the given image reference using ClamAV
+
+        Returns a tuple containing ClamAVs results and a string specifying the file which contained
+        the finding.
+        '''
+        logger.debug(f'scanning container image {image_reference}')
+        with tarfile.open(
```

Comment:
this is copy-paste from clamav.util. please do not copy-paste. refactor the function instead. either pass-in the scanning function, or yield the objects to scan from it